app/models/host/managed_extensions.rb

```diff
@@ -17,7 +17,10 @@ def queue_reboot
   end
 
   def setReboot
-    old.becomes(Host::Discovered).reboot
+    old.becomes(Host::Discovered).reboot if errors.empty?
```

Comment:
When provisioning a discovered node, we don't call hosts#create but discovered_hosts#create.